programs/compressed-pda/src/invoke/verify_state_proof.rs

```diff
@@ -103,14 +111,64 @@ pub fn hash_input_compressed_accounts<
                 // Vec::remove(addresses, j);
             }
         };
-
+        if current_mt_index
+            != input_compressed_account_with_context
+                .merkle_context
+                .merkle_tree_pubkey_index as i16
+        {
+            current_mt_index = input_compressed_account_with_context
+                .merkle_context
+                .merkle_tree_pubkey_index as i16;
+            let merkle_tree_pubkey = remaining_accounts[input_compressed_account_with_context
+                .merkle_context
+                .merkle_tree_pubkey_index
+                as usize]
+                .key();
+            current_hashed_mt = hash_to_bn254_field_size_be(&merkle_tree_pubkey.to_bytes())
+                .unwrap()
+                .0;
+            hashed_pubkeys.push((merkle_tree_pubkey, current_hashed_mt));
+        } else if current_mt_index
+            < input_compressed_account_with_context
+                .merkle_context
+                .merkle_tree_pubkey_index as i16
+        {
+            // TODO: add failing test
+            msg!("Invalid Merkle tree index: {} current index {} (Merkle tree indices need to be in ascendin order.", input_compressed_account_with_context
+            .merkle_context
+            .merkle_tree_pubkey_index as i16, current_mt_index);
+            return err!(CompressedPdaError::InvalidMerkleTreeIndex);
+        }
+        if owner_pubkey
+            != input_compressed_account_with_context
+                .compressed_account
+                .owner
+        {
+            owner_pubkey = input_compressed_account_with_context
+                .compressed_account
+                .owner;
+            hashed_owner = match hashed_pubkeys
+                .iter()
+                .find(|x| x.0 == inputs.output_compressed_accounts[j].owner)
+            {
+                Some(hashed_owner) => hashed_owner.1,
+                None => {
+                    let hashed_owner = hash_to_bn254_field_size_be(
+                        &inputs.output_compressed_accounts[j].owner.to_bytes(),
+                    )
+                    .unwrap()
+                    .0;
+                    hashed_pubkeys.push((inputs.output_compressed_accounts[j].owner, hashed_owner));
+                    hashed_owner
+                }
+            };
+        }
+        msg!("owner: {:?}", owner_pubkey);
```

Comment:
Is it necessary, or it was just a debug log?

Reply:
removed